internal/directives/argocd_updater.go

```diff
@@ -342,6 +344,23 @@ func (a *argocdUpdater) buildDesiredSources(
 					err,
 				)
 			}
+			if updateUsed {
+				continue updateLoop
+			}
+		}
+		if !updateUsed {
+			if srcUpdate.Chart == "" {
+				return nil, fmt.Errorf(
+					"no source of Argo CD Application %q in namespace %q matched update "+
+						"for source with repoURL %s",
+					app.Name, app.Namespace, srcUpdate.RepoURL,
+				)
+			}
+			return nil, fmt.Errorf(
+				"no source of Argo CD Application %q in namespace %q matched update "+
+					"for source with repoURL %s and chart %q",
+				app.Name, app.Namespace, srcUpdate.RepoURL, srcUpdate.Chart,
+			)
 		}
```

Comment:
If we looped through all sources, attempting to apply this update, and the update didn't match any of those sources, this is evidence of misconfiguration.